corehq/apps/app_manager/views/formdesigner.py

```diff
@@ -313,3 +172,201 @@ def get_form_data_schema(request, domain, form_unique_id):
 @require_GET
 def ping(request):
     return HttpResponse("pong")
+
+
+def _get_base_vellum_options(request, domain, app, displayLang):
+    """
+    Returns the base set of options that will be passed into Vellum
+    when it is initialized.
+    :param displayLang: --> derived from the base context
+    """
+    return {
+        'intents': {
+            'templates': next(app_callout_templates),
+        },
+        'javaRosa': {
+            'langs': app.langs,
+            'displayLanguage': displayLang,
+        },
+        'uploader': {
+            'uploadUrls': {
+                'image': reverse("hqmedia_uploader_image", args=[domain, app.id]),
+                'audio': reverse("hqmedia_uploader_audio", args=[domain, app.id]),
+                'video': reverse("hqmedia_uploader_video", args=[domain, app.id]),
+                'text': reverse("hqmedia_uploader_text", args=[domain, app.id]),
+            },
+            'objectMap': app.get_object_map(),
+            'sessionid': request.COOKIES.get('sessionid'),
+        },
+    }
+
+
+def _get_vellum_core_context(request, domain, app, module, form):
+    """
+    Returns the core context that will be passed into vellum when it is
+    initialized.
+    """
+    core = {
+        'dataSourcesEndpoint': reverse('get_form_data_schema',
+                                       kwargs={'domain': domain,
+                                               'form_unique_id': form.get_unique_id()}),
+        'dataSource': [
+            # DEPRECATED. Use dataSourcesEndpoint
```

Comment:
The entire 'dataSource': [...], (key and value) should be removed. Sorry for not making that more clear earlier.